File: AtCoder/C_T_shirts.go
package main

import "fmt"

func main() {
	var n, m int
	fmt.Scan(&n, &m)
	var s string
	fmt.Scan(&s)
	s += "0"

	mx, ones, twos := 0, 0, 0
	for i := 0; i <= n; i++ {
		if s[i] == '0' {
			mx = max(mx, twos + max(0, ones - m))
			ones = 0
			twos = 0
		} else if s[i] == '1' {
			ones++
		} else {
			twos++
		}
	}
	fmt.Println(mx)
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: AtCoder/C_T_shirts.rs
use proconio::input;
use std::cmp::max;

fn main() {
    input! {
        _n: usize,
        m: i32,
        mut s: String,
    }

    s += "0";

    let mut mx = 0;
    let mut ones = 0;
    let mut twos = 0;

    for c in s.chars() {
        match c {
            '0' => {
                mx = max(mx, twos + max(0, ones - m));
                ones = 0;
                twos = 0;
            }
            '1' => {
                ones += 1;
            }
            '2' => {
                twos += 1;
            }
            _ => (),
        }
    }

    println!("{}", mx);
}